pkg/agent/sysadvisor/plugin/inference/modelresultfetcher/borwein/borwein.go

```diff
@@ -0,0 +1,328 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package borwein
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	//nolint
+	"github.com/golang/protobuf/proto"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubelet/pkg/apis/resourceplugin/v1alpha1"
+
+	"github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/metacache"
+	"github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/plugin/inference/modelresultfetcher"
+	borweinconsts "github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/plugin/inference/models/borwein/consts"
+	borweininfsvc "github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/plugin/inference/models/borwein/inferencesvc"
+	borweintypes "github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/plugin/inference/models/borwein/types"
+	"github.com/kubewharf/katalyst-core/pkg/config"
+	"github.com/kubewharf/katalyst-core/pkg/config/generic"
+	"github.com/kubewharf/katalyst-core/pkg/consts"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver"
+	metricspool "github.com/kubewharf/katalyst-core/pkg/metrics/metrics-pool"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+	"github.com/kubewharf/katalyst-core/pkg/util/process"
+	"github.com/kubewharf/katalyst-core/pkg/util/qos"
+)
+
+const (
+	BorweinModelResultFetcherName = "borwein_model_result_fetcher"
+)
+
+type BorweinModelResultFetcher struct {
+	name      string
+	qosConfig *generic.QoSConfiguration
+
+	nodeFeatureNames              []string // handled by GetNodeFeature
+	containerFeatureNames         []string // handled by GetContainerFeature
+	inferenceServiceSocketAbsPath string
+
+	//metaServer *metaserver.MetaServer
+	//metaWriter metacache.MetaWriter
+	//metaReader metacache.MetaReader
+
+	infSvcClient borweininfsvc.InferenceServiceClient
+}
+
+const (
+	NodeFeatureNodeName = "node_feature_name"
+)
+
+type GetNodeFeatureValueFunc func(featureName string, metaServer *metaserver.MetaServer, metaReader metacache.MetaReader) (string, error)
+type GetContainerFeatureValueFunc func(podUID string, containerName string, featureName string,
+	metaServer *metaserver.MetaServer, metaReader metacache.MetaReader) (string, error)
+
+var GetNodeFeatureValue GetNodeFeatureValueFunc = NativeGetNodeFeatureValue
```

Comment:
1. GetNodeFeatureValue -> getNodeFeatureValue
2. should we lock the update actions for setter functions?

Reply:
since function type variable assignment is atomic in assembly code，maybe we don't need to lock setter functions currently.